handbook/engineering/search/index.md

```diff
@@ -2,9 +2,93 @@
 
 We own the end-to-end Sourcegraph search experience.
 
+## Vision
+
+Sourcegraph search is:
+
+- **Fast:** Performance of showing results is fast and I can get to what I’m looking for quickly.
+- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write.
+The answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
+- **Easy to use:** Users can figure out how to find what they are looking for, the syntax is clear, new users understand what options they have for searching.
+- **Sharable:** Sourcegraph search creates network effects that provide compounding value with scale. 
+
 ## Goals
 
-Our current goals are documented in the [search team's current tracking issue](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+_Updated 2020-08-27_
+
+## Long term (~1 year)
+
+Sourcegraph search is far and away the best code search solution, search results are fast, and I can find what I’m looking for quickly.
+
+## Medium term (3-6 months)
+
+In the next 6 months we want to be halfway to the long term goal. To achieve this, we need to accomplish the following outcomes:
+
+- Getting to search results faster through a better search experience.
+- What do we need to do to get to the speed of grep.app?
+- Scale fast search to 500k repositories.
+
+## Short term (1-3 months)
+
+Our current focus is documented in the [tracking issue for the current milestone](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aopen+is%3Aissue+label%3Atracking+label%3Ateam%2Fsearch).
+
+### Fast
+
+- **Streaming search**
+  - **Problem:** We want to make significant improvements to search that are blocked by not having the infrastructure needed.
+  - **Owner:** Keegan
+  - **Status:** In progress
+
+### Expressive
+
+- **Hierarchical search (AND/OR/NOT)**
+  - **Problem:** We have customers who want to migrate from OpenGrok and other search tools, and they want to be able to do searches that are available in those tools.
+  - **Outcomes:** Users can run complex searches on Sourcegraph.
```

Comment:
```suggestion
  - **Outcomes:** Users can express relations on code, files, repositories, and commits in search queries to more effectively filter the data they need
```

"Relations" here meaning, e.g., "repo has file" "repo has commit", ... etc.

Reply:
> "Relations" here meaning, e.g., "repo has file" "repo has commit", ... etc.

Hmm this is AND/OR search. Did you mean to suggest this for the goal below (Improve syntax for existing filters like `repohasfile`)?
> Sourcegraph is the only industrial-strength search solution that provides these capabilities.

This feels a bit vision-like or something we should aspire to be.